app/src/pages/home/homeSlice.ts

```diff
@@ -0,0 +1,325 @@
+    const res = await deleteComment({ ...data });
+    return {
+      commentId: data.commentId,
+      threadData: res,
+    };
+  }
+);
+
+// Feed thunks
+export const readHomeFeedLatestAsync = createAsyncThunk(
+  "home/readHomeFeedLatest",
+  async ({ query }: { query: string }) => {
+    const res = await getFeed({ query });
+    return res;
+  }
+);
+export const readHomeFeedNextAsync = createAsyncThunk(
+  "home/readHomeFeedNext",
+  async ({
+    oldestUpdateFromCurrentBucket,
+    latestUpdateFromNextNext,
+  }: {
+    oldestUpdateFromCurrentBucket: string;
+    latestUpdateFromNextNext: string;
+  }) => {
+    const query = `olderThanDate=${oldestUpdateFromCurrentBucket}&newerThanDate=${latestUpdateFromNextNext}`;
+    const res = await getFeed({ query });
+    return res;
+  }
+);
+
+export const homeSlice = createSlice({
+  name: "home",
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      // Thread reducers
+      .addCase(createThreadAsync.pending, (state) => {
+        stateStatus.loading(state, "uploading thread");
+      })
+      .addCase(createThreadAsync.fulfilled, (state, action) => {
+        stateStatus.idle(state);
+        state.threads = {
+          ...state.threads,
+          [action.payload.id]: action.payload,
+        };
+      })
+      .addCase(createThreadAsync.rejected, (state) => {
+        stateStatus.error(state, "unable to upload thread");
+      })
+      .addCase(readThreadsAsync.pending, (state) => {
+        stateStatus.loading(state, "getting thread");
+      })
+      .addCase(readThreadsAsync.fulfilled, (state, action) => {
+        stateStatus.idle(state);
+        state.threads = {
+          ...state.threads,
+          ...action.payload,
+        };
+      })
+      .addCase(readThreadsAsync.rejected, (state) => {
+        stateStatus.error(state, "unable to get thread");
+      })
+      // .addCase(updateThreadAsync.pending, (state) => {
+      //   stateStatus.loading(state, "updating thread");
+      // })
+      // .addCase(updateThreadAsync.fulfilled, (state, action) => {
+      //   stateStatus.idle(state);
+      //   state.threads = {
+      //     ...state.threads,
+      //     [action.payload.id]: action.payload,
+      //   };
+      // })
+      // .addCase(updateThreadAsync.rejected, (state) => {
+      //   stateStatus.error(state, "unable to update thread");
+      // })
+      // .addCase(deleteThreadAsync.pending, (state) => {
+      //   stateStatus.loading(state, "deleting thread");
+      // })
+      // .addCase(deleteThreadAsync.fulfilled, (state, action) => {
+      //   stateStatus.idle(state);
+      //   const newThreadsState = {...state.threads};
+      //   delete newThreadsState[action.payload.id];
+      //   state.threads = newThreadsState;
+      // })
+      // .addCase(deleteThreadAsync.rejected, (state) => {
+      //   stateStatus.error(state, "unable to delete thread");
+      // })
```

Comment:
Should we remove this?

Reply:
Let's leave them for now if you've no strong objections. The implementation for these is what it is so I don't think they're speculating (not far off anyway) on what they'll be when uncommented. They can be uncommented now without issue but they're easier to spot commented.